extension/url-handler.js

```diff
@@ -0,0 +1,98 @@
+
+var urlHandler = {};  // Namespace
+
+/**
+ * In most cases, we don't want to display a full URL with a path.
+ * @param {inputUrl} The original url.
+ * @returns A more minimized version of the url.
+ */
+urlHandler.GetMinimalUrl = function(inputUrl) {
+  var scheme = '';
+  var hostport = '';
+  var parser = document.createElement('a');
+  parser.href = inputUrl;
+
+  if (parser.protocol === 'chrome-extension:') {
+    // The parser interprets some urls without schemes as relative to
+    // chrome-extension:. If that happens, turn it into a https url instead.
+    parser.href = 'https://' + inputUrl;
+  }
+  if ((parser.protocol === 'file:' || parser.protocol === ':') && 
+      (inputUrl.substring(0, 6).indexOf(':') == -1)) {
+    // If the protocol is empty, it might be missing the scheme. Add a scheme
+    // and then try again.
+    inputUrl = 'http://' + inputUrl;
+    parser.href = inputUrl;
+  }
+
+  if (parser.protocol !== 'http:' && parser.protocol !== 'https:')
+    scheme = parser.protocol + '//';
+
+  return scheme + parser.host;
+};
+
+/**
+ * Check whether a URL corresponds to a domain known to produce a green lock.
+ * @param {String} url The url of interest. Should be a valid HTTPS url.
+ * @returns {bool} True if a url is from a green lock site.
+ */
+urlHandler.IsGreenLockSite = function(url) {
+  // Reject a site unless it's a domain known to produce a green lock.
+  var greenLockSites = [
+    '^google\\.\\\\*',
```

Comment:
There seems to be a bit too many escape characters here. How about converting each row to an actual JS Regexp object and just looping over the array? It'll be a bit slower since you'll now do the match muItiple times, but I think it will also save a couple of escape characters.

Like so:

[ /^google\..*/,
 /^facebook\.com/,
...
]